File: Hashmap/Two Sum/TwoSum.go
func twoSum(nums []int, target int) []int {
	d := make(map[int]int)
	var result []int

	for index, num := range nums {
		if idx, exists := d[target-num]; exists {
			result = append(result, idx, index)
			break
		}
		d[num] = index
	}
	return result
}

File: Hashmap/Two Sum/TwoSum.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        visited = {}
        for i in range(len(nums)):
            if target - nums[i] in visited:
                return [i, visited[target - nums[i]]]
            visited[nums[i]] = i

        # This should never happen as we will surely have an answer
        return []
